ethereumj-core/src/main/java/org/ethereum/mine/MinerListener.java

```diff
@@ -23,9 +23,30 @@
  * Created by Anton Nashatyrev on 10.12.2015.
  */
 public interface MinerListener {
+
+    enum MinerStatus {
```

Comment:
What is valuable for the user of that status? I suppose the most valuable things to be notified about are starting and ending of dataset creation. Concerning an option to mine with usage of light dataset only, this lifecycle not always looks perfect. I think that better to have following cycle: `LIGHT_STARTED` -> `FULL_STARTED (optionally)` -> `DATASET_GENERATED`. Last message could have two options `DATASET_GENERATED_LIGHT` and `DATASET_GENERATED_FULL`